server/service/osquery_utils/queries.go

```diff
@@ -164,6 +158,64 @@ var hostDetailQueries = map[string]DetailQuery{
 			return nil
 		},
 	},
+	"os_version_unix_like": {
+		// Collect platform-specific information to populate `host.OSVersion` for Darwin and Linux.
+		// Note that data for `operating_system` and `host_operating_system` tables are ingested via
+		// the `os_unix_like` extra detail query below.
+		Query:     `SELECT * FROM os_version LIMIT 1`,
+		Platforms: append(fleet.HostLinuxOSs, "darwin"),
+		IngestFunc: func(ctx context.Context, logger log.Logger, host *fleet.Host, rows []map[string]string) error {
+			if len(rows) != 1 {
+				logger.Log("component", "service", "method", "IngestFunc", "err",
+					fmt.Sprintf("detail_query_os_version_unix_like expected single result got %d", len(rows)))
+				return nil
+			}
+
+			host.OSVersion = fmt.Sprintf("%v %v", rows[0]["name"], parseOSVersion(
+				rows[0]["name"],
+				rows[0]["version"],
+				rows[0]["major"],
+				rows[0]["minor"],
+				rows[0]["patch"],
+				rows[0]["build"],
+			))
+
+			return nil
+		},
+	},
+	"os_version_windows": {
+		// Collect platform-specific information to populate `host.OSVersion` for Windows.
+		// Note that data for `operating_system` and `host_operating_system` tables are ingested via
+		// the `os_windows` extra detail query below.
+		Query: `SELECT
+			os.name,
+			r.data
+		FROM
+			os_version os,
+			(
+				SELECT
+					data
+				FROM
+					registry
+				WHERE
+					path = 'HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\DisplayVersion') r
+		LIMIT 1`,
+		Platforms: []string{"windows"},
+		IngestFunc: func(ctx context.Context, logger log.Logger, host *fleet.Host, rows []map[string]string) error {
+			if len(rows) != 1 {
+				logger.Log("component", "service", "method", "IngestFunc", "err",
+					fmt.Sprintf("detail_query_os_version_windows expected single result got %d", len(rows)))
+				return nil
+			}
+
+			s := fmt.Sprintf("%v %v", rows[0]["name"], rows[0]["data"])
+			// Shorten "Microsoft Windows" to "Windows" to facilitate display and sorting in UI
+			s = strings.Replace(s, "Microsoft Windows", "Windows", 1)
```

Comment:
I'd argue that this is a presentation concern and shouldn't be handled here.

Reply:
I thought so too at first. Then I realized that the frontend relies on server-side sort for this column and we'd need to be able to sort the values alphabetically on the backend and couldn't treat it as a purely presentational task for the frontend. It seemed like we might want to handle this translation at ingestion so sorting the hosts table by os_version would be simpler and performant. What do you think?

Reply:
in that case I'd modify the method that retrieves the data from the database, by either tweaking the SQL query or in go.

having that said, I think it's a good practice but I wouldn't die on that hill, I'm OK if you decide we should keep the logic as-is

Reply:
I totally agree about it being good practice in general. Would you have any performance concerns here about doing it at the db layer? The `ListHosts` query is quite complex already. It seems like we generally would want to avoid adding complexity there. Doing it in go when returning the `ListHosts` result also seems like it add to the response time (which I believe is already one of our longest running for the UI) because we'd need to be iterating over the results. I think the translation might need to be done in other retrieval methods that are returning host details so that seems like it could be less easy to maintain.

It seems like more and more the product is shifting from being a place where users go to see what osquery is reporting in its pure form to a place where we are combining and re-contextualizing that data. I wonder if it is time for a broader conversation about how we want to handle data transformations and normalization so that we can move away from our ad-hoc approach.

Reply:
that makes sense to me and seems like a fair compromise given that the queries are run periodically, so it won't be such a big deal if we want to remove this in the future (I think).